src/mon/Paxos.cc

```diff
@@ -1224,7 +1224,6 @@ void Paxos::lease_renew_timeout()
  */
 void Paxos::trim()
 {
-  assert(should_trim());
```

Comment:
i think this what `assert()` is for: to make something not that obvious more obvious.

Reply:
this is sort of gray area, but i think we can leave as it is unless it causes performance issue or it has some unpleasing side effects. `assert()` will be optimised out in future in NDEBUG build.

Reply:
ok